<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Filtering and analysis of MIP panel genetic datasets • mipmapper</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="jquery.sticky-kit.min.js"></script><script src="pkgdown.js"></script><meta property="og:title" content="Filtering and analysis of MIP panel genetic datasets">
<meta property="og:description" content="Molecular inversion probes (MIPs) are a tool for genomic
    partitioning, which result in datasets of SNP genotypes. mipmapper
    is a package to help assist in filtering the raw output from these
    sequencing efforts, and conducting population genetic analyses.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">mipmapper</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    

    
    
<div id="mipmapper" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#mipmapper" class="anchor"></a>mipmapper</h1></div>

<p>The R package <em>mipmapper</em> package contains a series of functions for analysing and visualising Molecular Inversion Probe (MIP) data. <strong>This package is in early stages of development</strong>, but will eventually include a range of methods for carrying out population genetic analyses. Full documentation can be found <a href="https://mrc-ide.github.io/mipmapper/">here</a>.</p>
<div id="installation" class="section level3">
<h3 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h3>
<p>In R, ensure that you have the devtools package installed by running</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">install.packages</span>(<span class="st">"devtools"</span>, <span class="dt">repos=</span><span class="st">'http://cran.us.r-project.org'</span>)</code></pre></div>
<p>Then we can simply install the <em>mipmapper</em> package directly from GitHub by running</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools::<span class="kw">install_github</span>(<span class="st">"mrc-ide/mipmapper"</span>)</code></pre></div>
<p>And we can load the package by running</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(mipmapper)</code></pre></div>
</div>
<div id="data-loading-and-filtering" class="section level3">
<h3 class="hasAnchor">
<a href="#data-loading-and-filtering" class="anchor"></a>Data loading and filtering</h3>
<p>Load raw data from .csv file. You will need to change the file path to where you have stored the data. An example of this is shown below, but commented out.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># if loading your own data, uncomment this line and change path to your data</span>
<span class="co"># dat0 &lt;- fast_read("path_to_your_data/NeutralSNPs_AheroYombo.csv")</span>

<span class="co"># here we will use in-built example data</span>
dat0 &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/mipmapper/topics/mipmapper_file">mipmapper_file</a></span>(<span class="st">"dummy_data.csv"</span>)</code></pre></div>
<p>Some miscellaneous filtering. Subset to SNPs only (i.e. no more complex mutations), group all alternative alleles together as a single “non-reference” allele, and drop irregular loci (for example non-integer barcode counts).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dat1 &lt;-<span class="st"> </span><span class="kw"><a href="reference/filter_misc.html">filter_misc</a></span>(dat0, <span class="dt">SNP_only =</span> <span class="ot">TRUE</span>, <span class="dt">group_Alt =</span> <span class="ot">TRUE</span>, <span class="dt">drop_irregular =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>Next we want to filter based on coverage, throwing away any loci that are below a minimum coverage level. We can visualise how much data will be left at different thresholds using the following function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/plot_coverage.html">plot_coverage</a></span>(dat1)</code></pre></div>
<p><img src="tools/README-plot_coverage-1.png" width="700"></p>
<p>Choose a threshold that strikes a balance between data quantity and quality. Once you have chosen a threshold, apply the filtering as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">my_threshold &lt;-<span class="st"> </span><span class="dv">6</span>
dat3 &lt;-<span class="st"> </span><span class="kw"><a href="reference/filter_coverage.html">filter_coverage</a></span>(dat1, <span class="dt">min_coverage =</span> my_threshold)</code></pre></div>
</div>
<div id="principal-components-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#principal-components-analysis" class="anchor"></a>Principal Components Analysis</h3>
<p>Before carry out PCA analysis we will convert our filtered dataset into a wide format, where each row is an unique sample, with new columns for each locus. This can be achieved as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dat4 &lt;-<span class="st"> </span><span class="kw"><a href="reference/melt_mip_data.html">melt_mip_data</a></span>(dat3)</code></pre></div>
<p>This can then be used to impute any missing values:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dat5 &lt;-<span class="st"> </span><span class="kw"><a href="reference/impute_mip_data.html">impute_mip_data</a></span>(dat4)</code></pre></div>
<p>The imputed data set can then be analysed using principal component analysis:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pca &lt;-<span class="st"> </span><span class="kw"><a href="reference/pca_mip_data.html">pca_mip_data</a></span>(dat5)</code></pre></div>
<p>We can view the variance explained by each compenet graphically using:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/plot_pca_variance.html">plot_pca_variance</a></span>(pca)</code></pre></div>
<div class="figure">
<img src="tools/pca_var.png">
</div>
<p>And lastly we can plot the actual prinical component analysis, to see how it has clustered our data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/plot_pca.html">plot_pca</a></span>(pca, <span class="dt">num_components =</span> <span class="dv">2</span>, <span class="dt">meta_var =</span> <span class="st">"Country"</span>)</code></pre></div>
<div class="figure">
<img src="tools/pca_2var.png">
</div>
<p>We can control whether we want to visualise the first 2 or 3 components, with the <code>num_componenets</code> argument:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/plot_pca.html">plot_pca</a></span>(pca, <span class="dt">num_components =</span> <span class="dv">3</span>, <span class="dt">meta_var =</span> <span class="st">"Country"</span>)</code></pre></div>
<div class="figure">
<img src="tools/pca_3var.png">
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Report a bug at <br><a href="https://github.com/mrc-ide/mipmapper/issues">https://​github.com/​mrc-ide/​mipmapper/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Bob Verity <br><small class="roles"> Author, maintainer </small>  </li>
<li>OJ Watson <br><small class="roles"> Author </small>  </li>
</ul>
</div>

      <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.org/mrc-ide/mipmapper"><img src="https://travis-ci.org/mrc-ide/mipmapper.svg?branch=master" alt="Travis build status"></a></li>
<li><a href="https://ci.appveyor.com/project/mrc-ide/mipmapper"><img src="https://ci.appveyor.com/api/projects/status/github/mrc-ide/mipmapper?branch=master&amp;svg=true" alt="AppVeyor build status"></a></li>
<li><a href="https://codecov.io/github/mrc-ide/mipmapper?branch=master"><img src="https://codecov.io/gh/mrc-ide/mipmapper/branch/master/graph/badge.svg" alt="Coverage status"></a></li>
</ul>
</div>
</div>

</div>


      <footer><div class="copyright">
  <p>Developed by Bob Verity, OJ Watson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
